src/Persistent_cohomology/test/persistent_cohomology_unit_test.cpp

```diff
@@ -146,9 +146,14 @@ void test_rips_persistence_in_dimension(int dimension) {
   std::clog << "str_rips_persistence=" << str_rips_persistence << std::endl;
 }
 
+BOOST_AUTO_TEST_CASE( rips_persistent_cohomology_single_field_dim_0 )
+{
+  BOOST_CHECK_THROW(test_rips_persistence_in_dimension(0), std::invalid_argument);
+}
+
 BOOST_AUTO_TEST_CASE( rips_persistent_cohomology_single_field_dim_1 )
 {
-  test_rips_persistence_in_dimension(1);
+  BOOST_CHECK_THROW(test_rips_persistence_in_dimension(1), std::invalid_argument);
```

Comment:
I find it problematic that this file confuses the dimension with the field of coefficients. It was like this before this patch so we can fix it separately later, but please open an issue so we don't forget.

Reply:
#515 has been opened